include/dr/mhp/views/zip.hpp

```diff
@@ -83,6 +83,31 @@ template <typename Base> auto base_to_segments(Base &&base) {
   return std::apply(bases_to_segments, base);
 }
 
+// based on https://ericniebler.github.io/range-v3/#autotoc_md30  "Create custom
+// iterators"
+template <typename Iter> struct cursor_over_local_ranges {
```

Comment:
It does look like a nice way to define an iterator. We would have to eliminate the range-v3 dependency.

Reply:
I may copy basic_iterator from ranges-v3 to our code base (change its namespace, remove not needed code from it, use it here). What do you think?

Reply:
It would be ok if you could isolate just this one thing without pulling the rest of ranges-v3